### ISSUE TITLE
Remove array decompression in this driver?

### ISSUE THREAD

**Issue by MarkRivers** [areaDetector/ADEiger#24]
Now that compressed NDArrays are supported in ADCore, this driver could export only compressed arrays, and not do the decompression.  

Those compressed arrays could then be fed directly to NDPluginPva and then to viewers. They could also be fed directly to NDFileHDF5 once it supports Direct Chunk Write. The arrays would not need to be first decompressed by ADEiger and then re-compressed by NDPluginPva and NDFileHDF5.

NDPluginCodec would then do decompression for other plugins (NDPluginStats, etc.) This requires adding lz4 and bitshuffle/lz4 compression (non-Blosc versions) to NDPluginCodec.  Depending on the application, NDPluginCodec might run at a reduced rate.

This will speed up the ADEiger driver, and spread the load over more cores.

**Comment by MarkRivers** [comment]
I have implemented a new parameter and a new record, StreamDecompress.  The allowed values are No (0) and Yes (1).  This flag tells the driver whether or not to decompress the data on the stream interface.

If StreamDecompress=Yes then the decompression is done in streamAPI::uncompress().  This handles lz4, bs32-lz4, and bs16-lz4.  For backwards compatibility this is the default the first time the new version of ADEiger is run. The value of StreamDecompress is in autosave so it remembers the selected value on restart.

If StreamDecompress=No then the compressed data is copied directly into the NDArray, and NDArray.codec and NDArray.compressedSize are set accordingly. NDPluginPva can serve these compressed arrays, for example to the ImageJ plugin. Other plugins will need to get their data from an NDPluginCodec plugin that decompresses the NDArrays.

The OPI screens have been updated to show this new record. 

I have tested the operation at both long exposure times where the data is 32-bits, and short exposure times where the data are 16-bits.
